src/yang/ietf-bgp.yang

```diff
@@ -858,6 +858,56 @@ module ietf-bgp {
             "Clear neighbors action.";
 
           input {
+            choice operation {
+              default operation-admin;
+              description
+                "The type of operation for the clear action.";
+              case operation-admin {
+                leaf admin {
+                  type empty;
+                  description
+                    "Closes the Established BGP session with a BGP
+                     NOTIFICATION message with the Administrative
+                     Reset error subcode.";
+                  reference
+                    "RFC 4486 - Subcodes for BGP Cease Notification
+                     Message.";
+                }
+              }
+              case operation-hard {
+                leaf hard {
+                  type empty;
+                  description
+                    "Closes the Established BGP session with a BGP
+                     NOTIFICATION message with the Hard Reset error
+                     subcode.";
+                  reference
+                    "RFC 8538, Section 3 - Notification Message
+                     Support for BGP Graceful Restart.";
+                }
+              }
+              case operation-soft {
+                leaf soft {
+                  type empty;
+                  description
+                    "Re-sends the current Adj-Rib-Out to this
+                     neighbor.";
+                }
+              }
+              case operation-soft-inbound {
+                leaf soft-inbound {
+                  // XXX JMH - if we wanted to permit future expansion
```

Comment:
Please remove XXX comment before committing the change, if you are happy with the explanation.